From a816321ff4c31b9db78c712094fba1159aa06ac6 Mon Sep 17 00:00:00 2001
From: kopal <kopal@broadcom.com>
Date: Thu, 4 Oct 2012 02:06:36 +0530
Subject: [PATCH 600/762] ethtool: added ethtool support.

Also updated defconfigs to enabled i2c and spi.

Based on Broadcom SDK 2.3.

Signed-off-by: kopal <kopal@broadcom.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/netlogic/common/Makefile |    2 +-
 arch/mips/netlogic/msgcfg/Makefile |    1 +
 arch/mips/netlogic/xlp/board.c     |    1 +
 drivers/i2c/busses/i2c-xlp.c       |    6 +++++-
 4 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/mips/netlogic/common/Makefile b/arch/mips/netlogic/common/Makefile
index 9fa7c3a..90e8109 100644
--- a/arch/mips/netlogic/common/Makefile
+++ b/arch/mips/netlogic/common/Makefile
@@ -2,7 +2,7 @@ EXTRA_CFLAGS := $(CFLAGS) -DNLM_HAL_LINUX_KERNEL -Iarch/mips/include/asm/netlogi
 obj-$(CONFIG_NLM_XLP) :=   cpu_proc.o memory.o bootinfo.o pic_hal.o pci_hal.o
 
 obj-$(CONFIG_NLM_XLP)			+= nlm_hal_sys.o nlm_evp_cpld.o nlm_hal_cpu_info.o
-obj-$(CONFIG_NLM_XLP)			+= nlm_hal.o fdt_helper.o
+obj-$(CONFIG_NLM_XLP)			+= nlm_hal.o nlm_eeprom.o fdt_helper.o
 obj-$(CONFIG_XLP_NAE_SUPPORT)		+= nlm_hal_mdio.o nlm_hal_vsemi_config.o nlm_hal_nae_interface.o
 obj-$(CONFIG_XLP_NAE_SUPPORT)		+= nlm_hal_nae.o 
 obj-$(CONFIG_XLP_FMN_SUPPORT)		+= nlm_hal_fmn_config.o
diff --git a/arch/mips/netlogic/msgcfg/Makefile b/arch/mips/netlogic/msgcfg/Makefile
index 2231346..770378b 100644
--- a/arch/mips/netlogic/msgcfg/Makefile
+++ b/arch/mips/netlogic/msgcfg/Makefile
@@ -17,6 +17,7 @@ ifneq ("$(HAL_DIR)", "")
 	$(Q)cp -fr $(HAL_DIR)/nlm_evp_cpld.c  ../common
 	$(Q)cp -fr $(HAL_DIR)/cortina_cs34x7  ../common
 	$(Q)cp -fr $(HAL_DIR)/nlm_hal_nae.c  ../common
+	$(Q)cp -fr $(HAL_DIR)/nlm_eeprom.c  ../common
 	$(Q)cp -fr $(HAL_DIR)/nlm_hal_vsemi_config.c ../common
 	$(Q)cp -fr $(HAL_DIR)/nlm_hal_cpu_info.c  ../common
 	$(Q)cp -fr $(FDT_DIR)/libfdt/contrib/fdt_helper.c  ../common
diff --git a/arch/mips/netlogic/xlp/board.c b/arch/mips/netlogic/xlp/board.c
index 8fc1610..0cdd4b6 100644
--- a/arch/mips/netlogic/xlp/board.c
+++ b/arch/mips/netlogic/xlp/board.c
@@ -32,6 +32,7 @@
 static struct i2c_board_info xlp_i2c_device_info[] __initdata = {
         {"ds1374",          0, 0x68, 0, 0, 0},
         {"max6657",         0, 0x4c, 0, 0, 0},
+        {"xlpge_eeprom",      0, 0x57, 0, 0, 0},
 };
 
 static int __init xlp_i2c_device_init(void)
diff --git a/drivers/i2c/busses/i2c-xlp.c b/drivers/i2c/busses/i2c-xlp.c
index a0cf2a6..8c7db1f 100644
--- a/drivers/i2c/busses/i2c-xlp.c
+++ b/drivers/i2c/busses/i2c-xlp.c
@@ -551,5 +551,9 @@ MODULE_AUTHOR("Netlogic MicroSystems");
 MODULE_DESCRIPTION("I2C adapter for XLP soc");
 MODULE_LICENSE("GPL");
 
-module_init (i2c_xlp_init);
+#ifdef MODULE
+module_init(i2c_xlp_init);
+#else
+arch_initcall(i2c_xlp_init);
+#endif
 module_exit (i2c_xlp_exit);
-- 
1.7.0.4

